test/test_opaque_obj_v2.py

```diff
@@ -292,6 +292,25 @@ def forward(self, arg0_1, arg1_1, arg2_1):
 
         _deregister_effectful_op("_TestOpaqueObject::noisy_inject")
 
+    def test_compile(self):
+        class Model(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+
+            def forward(self, rng_state, x):
+                x = torch.ops._TestOpaqueObject.noisy_inject(x, rng_state)
+                x = x * x
+                x = torch.ops._TestOpaqueObject.noisy_inject(x, rng_state)
+                x = x + x
+                return x
+
+        mod = Model()
+        rng = RNGState(0)
+        x = torch.ones(2, 3)
+
+        res = torch.compile(mod, fullgraph=True)(rng, x)
+        self.assertFalse(torch.allclose(res, x * x + x))
+
```

Comment:
1) can you do an assertexpectedinline on some graphs? preferably dynamo graph and aot graph?
2) we don't need to torch.compile a module. Can we do this test case without an nn.Module?
3) can you see what happens if we compile a function that captures the rng_state ? Dynamo should automatically lift the rng_state to being a graph input
4) what happens when we return the rng_state object? (i'd expect this to just work with no intervention from us)
5) calling a method on the rng_state object should induce a graph break, there should be a test here

Reply:
It might be good too to use an opaqueobject that is just a container around a tensor or an integer, so that we can check that compile didn't end up baking in some value too